torch/distributed/rpc/api.py

```diff
@@ -768,3 +769,29 @@ def rpc_async(to, func, args=None, kwargs=None, timeout=UNSET_RPC_TIMEOUT):
         >>> rpc.shutdown()
     """
     return _invoke_rpc(to, func, RPCExecMode.ASYNC, args, kwargs, timeout)
+
+def collect_all(futures):
```

Comment:
> Do we expect torch.futures to be used anywhere else than rpc?

Yes, we could replace `ProcessGroup::work` with that future. And there was also a discussion to let `loss.backward()` to return a `Future`. If that occurs, most likely it will return `torch.futures.Future`.

Reply:
Gotcha, yea in this case we should have these be in `torch.futures` so that the user doesn't have to import `torch.rpc` only to get access to `wait|collect_all()`

Reply:
Makes sense in retrospect, just moved to torch.futures package

Reply:
@jjlilley are you planning to remove `wait|collect_all()` from `rpc`? I see that you've moved the two functions to `futures` but for some reason I don't see them removed from `rpc`...

Reply:
Sorry about that! I actually removed them in my buffer, but my "C-x C-s" apparently didn't go through. Uploading a new version

Reply:
ah that emacs... :)